lrpar/src/lib/ctbuilder.rs

```diff
@@ -1042,161 +1042,164 @@ where
-                            ridx = usize::from(ridx),
-                            pidx = usize::from(pidx),
-                            parse_paramname = parse_paramname,
-                            args = args.join(", ")).ok();
+                        quote!{
+                            #actionskind::#actionsvariant(#action_fn(#ridx_var, #lexer_var, #span_var, #parse_paramname, #(#args,)*))
+                        }
                     }
-                }
+                })
             } else if pidx == grm.start_prod() {
                 // The action for the start production (i.e. the extra rule/production
                 // added by lrpar) will never be executed, so a dummy function is all
                 // that's required. We add "unreachable" as a check in case some other
                 // detail of lrpar changes in the future.
-                if parse_paramname != "()" {
+                if !parse_param_unit {
                     // If the parse parameter is the unit type, `let _ = ();` leads to Clippy
                     // warnings.
-                    write!(outs, "\n        let _ = {parse_paramname:};").ok();
+                    //
+                    // As of rust 1.85.0 I don't see that lint triggering anymore, it might
+                    // have been relaxed since .
+                    wrapper_fn_body.extend(quote! {
+                        // This appears to be here to silence "unused_variables" lint,
+                        // in a way that doesn't require renaming the variable.
+                        let _ = #parse_paramname;
+                    });
                 }
-                outs.push_str("\n        unreachable!()");
+                wrapper_fn_body.extend(quote!(unreachable!()));
             } else {
                 panic!(
                     "Production in rule '{}' must have an action body.",
                     grm.rule_name_str(grm.prod_to_rule(pidx))
                 );
-            }
-            outs.push_str("\n    }\n\n");
-        }
-
-        // Wrappers enum
+            };
 
-        write!(
-            outs,
-            "    #[allow(dead_code)]
-    enum {}<'input> {{\n",
-            ACTIONS_KIND
-        )
-        .ok();
+            // TODO we can emit #[allow(unused_variables)] when pidx == grm.start_prod
+            // That should allow us to remove parse_param_unit and the associated branches.
+            wrappers.extend(quote!{
+                fn #wrapper_fn<'lexer, 'input: 'lexer>(
+                    #ridx_var: ::cfgrammar::RIdx<#storaget>,
+                    #lexer_var: &'lexer dyn ::lrpar::NonStreamingLexer<'input, #lexertypest>,
+                    #span_var: ::cfgrammar::Span,
+                    mut #args_var: ::std::vec::Drain<#edition_lifetime ::lrpar::parser::AStackType<<#lexertypest as ::lrpar::LexerTypes>::LexemeT, #actionskind<'input>>>,
+                    #parse_paramdef
+                ) -> #actionskind<'input> {
+
```

Comment:
I guess the blank line is unintentional? If so, please force push an update.

Reply:
Indeed, pushed.